client/src/components/WeatherForm.jsx

```diff
@@ -0,0 +1,87 @@
+import React, { useState } from "react";
+import WeatherCard from "./WeatherCard";
+import {
+  CircularProgress,
+  TextField,
+  Button,
+  Container,
+  Paper,
+} from "@mui/material";
```

Comment:
Pro MUI tip - specify a default import statement per MUI component instead of destructuring them all in one import statement like this. It's because by doing it the way you have, you're wastefully importing ALL of the MUI components but then only extracting the 5 of them above for use and discarding the rest. This leads to slower performance. Usually importing non-default modules is totally fine (such as `{ useState } from "react"`), but MUI is a particular exception due to it's larger size of the data it includes. Read the first half of this doc up through Option 1 (no need to read Option 2): https://mui.com/material-ui/guides/minimizing-bundle-size/

See how I do it for my app: https://github.com/jason102/fav-map-locations/blob/feature/code-for-review/client/src/pages/logged-in-pages/Location/index.tsx